tensorflow/python/distribute/mirrored_strategy.py

```diff
@@ -399,7 +399,7 @@ def _input_workers_with_options(self, options=None):
       return input_lib.InputWorkers(self._input_workers_devices)
     if options.experimental_replication_mode == distribute_lib.InputReplicationMode.PER_REPLICA:
       self._input_workers_devices = (
-          tuple((device_util.canonicalize("/device:CPU:0", d), (d,)) for d in self._devices))
```

Comment:
You have missed the other case here. This code is fine if `place_dataset_to_device=True` tht works for DALI. you need the previous case still when `place_dataset_to_device=False`. Please add that back.

Reply:
Added the other case as well.